api_v3/services/StorageProfileService.php

```diff
@@ -193,17 +193,19 @@ function lockPendingFileSyncsAction(KalturaFileSyncFilter $filter, $workerId, $s
 			$idLimit = min($lastId + self::MAX_FILESYNC_ID_RANGE, $maxId);
 			$fileSyncs = FileSync::getFileSyncsChunkNoCriteria($baseCriteria, $lastId, $idLimit);
 
-			if (!$fileSyncs)
+			if (count($fileSyncs) < KalturaFileSyncFilter::MAX_FILESYNCS_PER_CHUNK)
 			{
-				break;
+				$lastId = $idLimit;
 			}
-
-			if (count($fileSyncs) < KalturaFileSyncFilter::MAX_FILESYNCS_PER_CHUNK)
+			else
 			{
-				$done = true;
+				$lastId = end($fileSyncs)->getId();
```

Comment:
$done is useless - need to remove from the while()

Reply:
There is one use in $done.
The query does not contain ''createdAtLessThanOrEqual".
We check that later in 'filterFileSyncs'.
if we find file syncs that are valid for this filter, we don't want to continue in the loop, but the lastID will be smaller than maxId.
Hence the $done.